src/ostree/ot-admin-builtin-upgrade.c

```diff
@@ -104,6 +121,21 @@ ot_admin_builtin_upgrade (int argc, char **argv, GCancellable *cancellable, GErr
         }
     }
 
+  /* In the pull-only case, we do a cleanup here to ensure that if multiple
+   * commits were pulled, we garbage collect any intermediate data before
+   * pulling more.  This is really a best practice in general, but for maximum
+   * compatiblity, we only do cleanup if a user specifies the new --pull-only
+   * option.  Otherwise, we would break the case of trying to deploy a commit
+   * that isn't directly referenced.
+   */
```

Comment:
If a pull is successful, it will set the ref.  Let's call this the "new pulled base", as distinct from creating a new deployment.  `ostree_sysroot_cleanup` won't reset the ref.

Hence we should (though this isn't explicitly tested right now) keep deltas, since we still have the ref.  Now, that's all the argument for why this won't break anything.

Why I wanted to do this is - think about a scenario where due to a flaky internet connection we do a pull, it gets interrupted, do another pull, it gets interrupted, repeat.  If this happens a lot, we will endlessly accumulate partial new commits.

With this change, we'll trigger the one-day-timeout on staged commit data.  Now...perhaps we should really do the cleanup *after* an unsuccessful pull, to ensure that we will always reuse downloaded-but-not-consumed deltas?  Will look at that.